gnucash/gnucash-bin.c

```diff
@@ -600,6 +610,72 @@ get_file_to_load()
         return gnc_history_get_last();
 }
 
+static void
+inner_main_run_report(void *closure, int argc, char **argv)
+{
+    SCM scm_report_output;
+    QofSession *session = NULL;
+
+    scm_c_eval_string("(debug-set! stack 200000)");
+    scm_set_current_module (scm_c_resolve_module ("gnucash utilities"));
+    scm_c_use_module("gnucash app-utils");
+
+    scm_c_use_module("gnucash reports");
+    scm_c_use_module("gnucash report-menus");
+
+    /* scm_set_current_module (scm_c_resolve_module ("gnucash report")); */
+    /* scm_set_current_module (scm_c_resolve_module ("gnucash reports")); */
+    gnucash_register_add_cell_types ();
+    gnc_report_init ();
+    load_gnucash_plugins();
+    load_gnucash_modules();
+    load_system_config();
+    load_user_config();
```

Comment:
I think this does way too much. Only load what you need.
I doubt you need to load_gnucash_modules. The only modules left are the optional ones (ofx, python and aqbanking). You don't need those to generate a report. In the same way you don't need any of the plugins loaded with load_gnucash_plugins. They too are related to importers and log-replay. Generating a report doesn't import and won't change transactions.